src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php

```diff
@@ -38,6 +38,7 @@ public function process(ContainerBuilder $container)
                 $definition->setFile($parameterBag->resolveValue($definition->getFile()));
                 $definition->setArguments($parameterBag->resolveValue($definition->getArguments()));
                 $definition->setFactoryClass($parameterBag->resolveValue($definition->getFactoryClass()));
+                $definition->setFactory($parameterBag->resolveValue($definition->getFactory()));
```

Comment:
Of you did it this way, parameters would also be responded in factory method names (see #13924 for another approach).